logoutresponse.go

```diff
@@ -0,0 +1,250 @@
+package saml
+
+import (
+	"encoding/base64"
+	"encoding/xml"
+	"errors"
+	"time"
+
+	"github.com/dorsha/go-saml/util"
+)
+
+func ParseCompressedEncodedLogoutResponse(b64ResponseXML string) (*LogoutResponse, error) {
+	logoutresponse := LogoutResponse{}
+	compressedXML, err := base64.StdEncoding.DecodeString(b64ResponseXML)
+	if err != nil {
+		return nil, err
+	}
+	bXML := util.Decompress(compressedXML)
+	err = xml.Unmarshal(bXML, &logoutresponse)
+	if err != nil {
+		return nil, err
+	}
+
+	// There is a bug with XML namespaces in Go that's causing XML attributes with colons to not be roundtrip
+	// marshal and unmarshaled so we'll keep the original string around for validation.
+	logoutresponse.originalString = string(bXML)
+	return &logoutresponse, nil
+
+}
+
+func ParseEncodedLogoutResponse(b64ResponseXML string) (*LogoutResponse, error) {
+	bytesXML, err := base64.StdEncoding.DecodeString(b64ResponseXML)
+	if err != nil {
+		return nil, err
+	}
+	return ParseDecodedLogoutResponse(bytesXML)
+}
+
+func ParseDecodedLogoutResponse(responseXML []byte) (*LogoutResponse, error) {
+	response := LogoutResponse{}
+	err := xml.Unmarshal(responseXML, &response)
+	if err != nil {
+		return nil, err
+	}
+	// save the original response because XML Signatures are fussy
+	response.originalString = string(responseXML)
+	return &response, nil
+}
+
+func (r *LogoutResponse) Validate(s *ServiceProviderSettings) error {
```

Comment:
We will need to check the validation with serval Identify Providers (ADFS, PingFederate, ..)

Reply:
Agreed